databroker/tests/test_broker.py

```diff
@@ -1262,3 +1263,62 @@ def test_monitoring(db, RE, hw):
     sd.monitors.append(hw.rand)
     RE(bp.count([hw.det], 5, delay=1))
     assert len(db[-1].table('rand_monitor')) > 1
+
+
+@py3
+def test_interlace_gens():
+    from databroker.eventsource.shim import interlace_gens
+    a = ({'time': i} for i in range(10) if i % 2 == 0)
+    b = ({'time': i} for i in range(10) if i % 2 == 1)
+    c = ({'time': i} for i in range(100, 110))
+    d = interlace_gens(a, b, c)
+    expected = list(range(10)) + list(range(100, 110))
+    for z, zz in zip(d, expected):
+        assert z['time'] == zz
+
+
+@py3
+def test_order(db, RE, hw):
+    from ophyd import sim
+    RE.subscribe(db.insert)
+    d = sim.SynPeriodicSignal(name='d', period=.5)
+    uid, = RE(monitor_during_wrapper(count([hw.det], num=7, delay=0.1), [d]))
+
+    t0 = None
+    for name, doc in db[uid].documents():
+        print(name, doc['time'])
+        # TODO: include datums in here at some point
+        if name in ['event']:
+            t1 = doc['time']
+            if t0:
+                assert t1 > t0
+            t0 = t1
+
+@py3
+def test_res_datum(db, RE, hw):
+    from ophyd.sim import NumpySeqHandler
+    import copy
+    db.prepare_hook = lambda name, doc: copy.deepcopy(doc)
+    for spec in NumpySeqHandler.specs:
+        db.reg.register_handler(spec, NumpySeqHandler)
+    L = []
+    RE.subscribe(db.insert)
+    RE.subscribe(lambda *x: L.append(x))
+
+    uid, = RE(count([hw.img], num=7, delay=0.1))
+
+    names = set()
+    for (n1, d1), (n2, d2) in zip(db[uid].documents(), L):
+        print(n1)
+        names.add(n1)
+        assert n1 == n2
+        # It seems that some of the documents don't have key parity
+        if n1 == 'resource':
+            d1.pop('id')
+            d2.pop('run_start')
```

Comment:
if 'run_start' is not coming back out of the databroker than is a bug someplace else..

Reply:
On my local sqlite and hdf5 don't have a `run_start` with their resources.

Reply:
That is a bug in those asset registries, does it work with mongo?

Reply:
It does work with mongo (on my local)

Reply:
example: https://travis-ci.org/NSLS-II/databroker/jobs/467728070#L1943

Reply:
It seems that the sqlite asset registry doesn't have a column for `start_doc`

Reply:
@tacaswell and I are both on board with changing the schema for this. Does that work for you?

Reply:
Maybe, I don't know what that means in concrete terms, though.